includes/api/ApiQueryUserContributions.php

```diff
@@ -225,14 +234,14 @@ private function prepareQuery() {
 				// STRAIGHT_JOIN
 				$tables = array( 'revision', 'recentchanges', 'page' );
 				$this->addOption( 'STRAIGHT_JOIN' );
-				$this->addWhere( 'rc_user_text=rev_user_text' );
+				$this->addWhere( '(rc_user != 0 AND rc_user=rev_user OR rc_user_text=rev_user_text)' );
```

Comment:
should we add an extra set of parentheses here?

Reply:
it should work properly, but I'll add them just for better readability

Reply:
`AND` has precedence over `OR`, so `x AND y OR z` is identical to `(x AND y) OR z`. `STRAIGHT JOIN` will behave quite differently than `LEFT JOIN`, as it forces the left table always be read before the right table.  Essentially will produce the same result as `RIGHT JOIN` with `rc_user != 0` moved to `WHERE`, so all anonymous entries will not be present in the result.

Compare the three examples and verify that you will be getting the same results as you would get before he change:

```
mysql@geo-db-dev-db-slave.query.consul[wikicities]>SELECT wl_namespace, wl_title, user_name FROM watchlist STRAIGHT JOIN user ON ( wl_namespace != 0 AND wl_namespace = user_id ) ORDER BY wl_wikia_addedtimestamp ASC LIMIT 10;
+--------------+---------------------------------------------------------------------+----------------+
| wl_namespace | wl_title                                                            | user_name      |
+--------------+---------------------------------------------------------------------+----------------+
|            1 |                                                                     | W-Jasonr       |
|            6 | Cu.JPG                                                              | Motorcitymagic |
|            7 | Cu.JPG                                                              | Mike           |
|          500 | Crucially/Explanation_of_the_Sunday_outage                          | Alien          |
|          500 | Crucially/Iowa_datacenter_outage                                    | Alien          |
|          500 | Crucially/Iowa_datacenter_test                                      | Alien          |
|          500 | IB_Clever/IB_Theory_Of_Knowledge_(TOK)                              | Alien          |
|          500 | Kacieh/Win_a_Trip_on_the_Glee_Wiki                                  | Alien          |
|          500 | Kirkburn/Beta_test_the_new_rich_text_editor                         | Alien          |
|          500 | Kirkburn/Finding_the_latest_activity_on_your_wiki_just_got_simpler! | Alien          |
+--------------+---------------------------------------------------------------------+----------------+
10 rows in set (0.00 sec)

mysql@geo-db-dev-db-slave.query.consul[wikicities]>SELECT wl_namespace, wl_title, user_name FROM watchlist RIGHT JOIN user ON ( wl_namespace = user_id ) WHERE wl_namespace != 0 ORDER BY wl_wikia_addedtimestamp ASC LIMIT 10;
+--------------+---------------------------------------------------------------------+----------------+
| wl_namespace | wl_title                                                            | user_name      |
+--------------+---------------------------------------------------------------------+----------------+
|            1 |                                                                     | W-Jasonr       |
|            6 | Cu.JPG                                                              | Motorcitymagic |
|            7 | Cu.JPG                                                              | Mike           |
|          500 | Crucially/Explanation_of_the_Sunday_outage                          | Alien          |
|          500 | Crucially/Iowa_datacenter_outage                                    | Alien          |
|          500 | Crucially/Iowa_datacenter_test                                      | Alien          |
|          500 | IB_Clever/IB_Theory_Of_Knowledge_(TOK)                              | Alien          |
|          500 | Kacieh/Win_a_Trip_on_the_Glee_Wiki                                  | Alien          |
|          500 | Kirkburn/Beta_test_the_new_rich_text_editor                         | Alien          |
|          500 | Kirkburn/Finding_the_latest_activity_on_your_wiki_just_got_simpler! | Alien          |
+--------------+---------------------------------------------------------------------+----------------+
10 rows in set (0.00 sec)

mysql@geo-db-dev-db-slave.query.consul[wikicities]>SELECT wl_namespace, wl_title, user_name FROM watchlist LEFT JOIN user ON ( wl_namespace != 0 AND wl_namespace = user_id ) ORDER BY wl_wikia_addedtimestamp ASC LIMIT 10;
+--------------+---------------------------------------------+----------------+
| wl_namespace | wl_title                                    | user_name      |
+--------------+---------------------------------------------+----------------+
|            0 |                                             | NULL           |
|            1 |                                             | W-Jasonr       |
|            6 | Cu.JPG                                      | Motorcitymagic |
|            7 | Cu.JPG                                      | Mike           |
|          500 | Crucially/Explanation_of_the_Sunday_outage  | Alien          |
|          500 | Crucially/Iowa_datacenter_outage            | Alien          |
|          500 | Crucially/Iowa_datacenter_test              | Alien          |
|          500 | IB_Clever/IB_Theory_Of_Knowledge_(TOK)      | Alien          |
|          500 | Kacieh/Win_a_Trip_on_the_Glee_Wiki          | Alien          |
|          500 | Kirkburn/Beta_test_the_new_rich_text_editor | Alien          |
+--------------+---------------------------------------------+----------------+
10 rows in set (0.01 sec)
```